### ISSUE TITLE
Config option "allowAdminChanges" which only allows editing the configuration in local development

### ISSUE THREAD

**Issue by mandrasch** [statamic/ideas#828]
Hey everyone,

tl;dr: Would love to have CraftCMSes [allowAdminChanges](https://craftcms.com/docs/3.x/config/config-settings.html#allowadminchanges) config option in Statamic. Editing blueprints, collection data structure, etc. would only possible in local development with this option, not on production sites dashboard.

### Background

I fully understand that the [Git Automation](https://statamic.dev/git-automation) is a handy feature for many statamic devs and I guess a lot of hard work was put into this feature. 

Personally I'm not a fan of having the git repository on the production server and push changes back to the repository.
_(Mainly because I do not want to fiddle around with merge conflicts on the command line on production. Had this too many times on other PHP projects 😫.)_

I rather would love

-  **to change my code / configuration / templates in local development, submit it via git and let CI/CD pipelines (Github/Gitlab/Bitbucket actions) take care of deploying.** See e.g. this [typo3 CI/CD deployment approach](https://t3terminal.com/blog/typo3-gitlab-deployment/) as example:

![image](https://user-images.githubusercontent.com/777278/179351577-ff1ac897-54cb-4211-89f6-9e5bfb6d2bf9.png)

- **On the production site, admins and editors should only be allowed to change the content in the Statamic dashboard, not change the configuration (blueprints, data structure, etc.)** 

- **The production server never pushes changes back to the git repository.**

But  how do I test locally with latest content?

For my local development I use the open source docker tool [DDEV](https://github.com/drud/ddev), which has workflows for pulling files and folders via SSH/rsync from production. Therefore I could just run `ddev pull production` to get the latest `content/` and `media assets` from production server to my local development and test my code changes with the content. This workflow could be used with Sail, Valet, etc. as well I guess. _(See [Statamic+DDEV](https://dev.to/mandrasch/setup-statamic-via-ddev-39bi) for more info)_

(The pulled / downloaded content is gitignored of course)

### Idea description

I found out that CraftCMS offers the workflow I would love to use in Statamic as well, they describe it this way ([Source](https://craftcms.com/knowledge-base/deployment-best-practices#commit-template-amp-config-changes)):

> **Deployment Best Practices**
> It’s best to limit config and dependency changes to local environments where they can be tested, committing explicit records of those changes to be used by the deployment process:
> [Set up project config](https://craftcms.com/docs/3.x/project-config.html#environment-setup) and [disable admin changes in production](https://craftcms.com/docs/3.x/config/config-settings.html#allowadminchanges) so Craft config changes are only made in development environments and committed from the config/project/ directory.
Only run composer update locally and commit updates to composer.lock after testing.

**CraftCMS has an option called [allowAdminChanges](
https://craftcms.com/docs/3.x/config/config-settings.html#allowadminchanges):**

> Whether admins should be allowed to make administrative changes to the system.
> When this is disabled, the Settings section will be hidden, the Craft edition and Craft/plugin versions will be locked, and the project config and Plugin Store will become read-only—though Craft and plugin licenses may still be purchased.

Since I'm a Statamic newbie, I'm not quite sure yet which implications such an option would have. 

**I drafted a quick implementation idea here: https://github.com/statamic/cms/discussions/6289**. Basically all permissions beginning with "Configure ..." must be deactivated on production.

```php
// vendor/statamic/cms/src/Auth/File/User.php
 
// Throw out all "Configure ..." permissions on production site
 if(env('APP_ENV') === 'production'){
            foreach ($permissions as $index => $permission) {
                if (strpos($permission, 'configure') !== false) {
                    unset($permissions[$index]);
                }
                if($permission == 'super'){
                    unset($permissions[$index]);
                }
          }
}
```

It could be based on `APP_ENV=production/local` or be a new setting in `.env`.

### Potentials for Statamic

In my personal opinion offering this deployment approach as well (next to Git Automation) could be a huge step forward for Statamic. Heard on twitter that there are some people who doesn't like the Git Automation approach as well. Therefore this config option could make Statamic more attractive for those group of people and broaden the possibilites. 

Happy to hear your feedback or questions! 👋

**Comment by jasonvarga** [comment]
Closing this in favor of the discussion you already have open on the cms repo.

https://github.com/statamic/cms/discussions/6289
